### PR TITLE
[extractor/iwara:user] Make paging better

### PR DESCRIPTION
**Pull request by Lesmiscore**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

---

### Description of your *pull request* and other information

Iwara has some edge-cases that user profile URL and video listing URL are slightly different. This PR fixes to work flawlessly with these cases.

Related: ytdl-patched/ytdl-patched#34

### CODE REVIEW

File: yt_dlp/extractor/iwara.py [hunk 1]
```diff
@@ -171,37 +172,78 @@ class IwaraUserIE(IwaraBaseIE):
...
-
-        page_urls = re.findall(
-            r'class="pager-item"[^>]*>\s*<a[^<]+href="([^"]+)', webpage)
-
-        for n, path in enumerate(page_urls, 2):
+    def _entries(self, playlist_id, base_url):
+        # we first probe users' profile page for existence of More videos page
+        webpage = self._download_webpage(
+            f'{base_url}/users/{playlist_id}', playlist_id)
+        videos_url = self._search_regex(r'<a href="(/users/[^/]+/videos)(?:\?[^"]*)?">', webpage, 'all videos url', default=None)
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:224]:
```suggestion
        videos_url = self._search_regex(r'<a href="(/users/[^/]+/videos)(?:\?[^"]*+)?">', webpage, 'all videos url', default=None)
```

**Review by pukkandan — CHANGES_REQUESTED** [review]
the comments are useless. Otherwise lgtm

File: yt_dlp/extractor/iwara.py (hunk 1 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/iwara.py]: 👎 https://github.com/ytdl-patched/ytdl-patched/blob/ytdlp/yt_dlp/extractor/iwara.py#L182

Comment (afterdelight) [reply yt_dlp/extractor/iwara.py]: what?

**Pull request merged**